### ISSUE TITLE
Unable to load plugin "store.smx": Native "Opts_GetOptBool" was not found

### ISSUE THREAD

**Issue by epicbluffer** [Kxnrl/Store#88]
I installed Store SM1.10 latest build, and follow the instructions, everything is quite sure until the last step about databases.cfg

Here is the error log:
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "store.smx": Native "Opts_GetOptBool" was not found
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_givecreditscommand.smx": Could not find required plugin "store"
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_pet.smx": Could not find required plugin "store"
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_musickit.smx": Could not find required plugin "store"
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_weaponskin.smx": Could not find required plugin "store"
L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_defaultskin.smx": Could not find required plugin "store"

And here is what inside databases.cfg :
"Databases"
{
	"driver_default"		"mysql"
	
	// When specifying "host", you may use an IP address, a hostname, or a socket file path
	
	"default"
	{
		"driver"			"default"
		"host"				"localhost"
		"database"			"sourcemod"
		"user"				"root"
		"pass"				""
		//"timeout"			"0"
		//"port"			"0"
	}
	
	"storage-local"
	{
		"driver"			"sqlite"
		"database"			"sourcemod-local"
	}

	"clientprefs"
	{
		"driver"			"sqlite"
		"host"				"localhost"
		"database"			"clientprefs-sqlite"
		"user"				"root"
		"pass"				""
		//"timeout"			"0"
		//"port"			"0"
	}
"csgo"
{
    "driver"    "mysql" // mysql support only
    "host"      "45.137.19.28"
    "database"  "storeze"
    "user"      "root"
    "pass"      "xxxxxxxx"
    "port"      "27015"
}
}

Its my first time to setup a server, but the importing sql to database part should be good. And its a zombie escape server so I delete all the store_gamemode.smx except store_ze and renamed it into store.smx. What should I do to fix this problem? Thank you

**Comment by epicbluffer** [comment]
> I installed Store SM1.10 latest build, and follow the instructions, everything is quite sure until the last step about databases.cfg
> 
> Here is the error log:
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "store.smx": Native "Opts_GetOptBool" was not found
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_givecreditscommand.smx": Could not find required plugin "store"
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_pet.smx": Could not find required plugin "store"
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_musickit.smx": Could not find required plugin "store"
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_weaponskin.smx": Could not find required plugin "store"
> L 10/17/2019 - 05:11:16: [SM] Unable to load plugin "modules/store_defaultskin.smx": Could not find required plugin "store"
> 
> And here is what inside databases.cfg :
> "Databases"
> {
> "driver_default" "mysql"
> 
> ```
> // When specifying "host", you may use an IP address, a hostname, or a socket file path
> 
> "default"
> {
> 	"driver"			"default"
> 	"host"				"localhost"
> 	"database"			"sourcemod"
> 	"user"				"root"
> 	"pass"				""
> 	//"timeout"			"0"
> 	//"port"			"0"
> }
> 
> "storage-local"
> {
> 	"driver"			"sqlite"
> 	"database"			"sourcemod-local"
> }
> 
> "clientprefs"
> {
> 	"driver"			"sqlite"
> 	"host"				"localhost"
> 	"database"			"clientprefs-sqlite"
> 	"user"				"root"
> 	"pass"				""
> 	//"timeout"			"0"
> 	//"port"			"0"
> }
> ```
> 
> "csgo"
> {
> "driver" "mysql" // mysql support only
> "host" "45.137.19.28"
> "database" "storeze"
> "user" "root"
> "pass" "xxxxxxxx"
> "port" "27015"
> }
> }
> 
> Its my first time to setup a server, but the importing sql to database part should be good. And its a zombie escape server so I delete all the store_gamemode.smx except store_ze and renamed it into store.smx. What should I do to fix this problem? Thank you

**Comment by Kxnrl** [comment]
I am checking that.

**Comment by Kxnrl** [comment]
try this https://build.kxnrl.com/Store/Store-SM1.10-latest.7z

**Comment by epicbluffer** [comment]
> try this https://build.kxnrl.com/Store/Store-SM1.10-latest.7z

Thanks but its still the same issue

**Comment by Kxnrl** [comment]
look like you downloaded that CDN cache file.

try this
https://build.kxnrl.com/Store/1.10/538-master-ee6350f.7z

**Comment by epicbluffer** [comment]
> look like you downloaded that CDN cache file.
> 
> try this
> https://build.kxnrl.com/Store/1.10/538-master-ee6350f.7z

Issue fixed, Thanks!!

**Comment by epicbluffer** [comment]
> > look like you downloaded that CDN cache file.
> > try this
> > https://build.kxnrl.com/Store/1.10/538-master-ee6350f.7z
> 
> Issue fixed, Thanks!!

Btw May I ask after the upload of the resources, it says
"[Store]  Please wait. Your inventory hasn't been fetched yet." for quite a while, is it I have to wait or somethings wrong with my database?

**Comment by Kxnrl** [comment]
You can check errlogs in {sourcemod path}/logs/errlog_{date}.log

**Comment by epicbluffer** [comment]
> You can check errlogs in
> {sourcemod path}/logs/errlog_{date}.log
> and
> {sourcemod path}/logs/store.error.log

Error log indicated this:
L 10/17/2019 - 06:55:49: Failed to connect to SQL database. [005] Error: [2002]: Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)

I tried to figure whats wrong, but cant fix yet.
I am using CentOS 7 64 bit and using MariaDB 10.2, when I type mysql in command prompt, it bring me into the MySQL shell prompt; So i guess my mysql server is running fine? 

In my databases.cfg, I set up like this:
"Databases"
{
	"driver_default"		"mysql"
	
	// When specifying "host", you may use an IP address, a hostname, or a socket file path
	
	"default"
	{
		"driver"			"default"
		"host"				"localhost"
		"database"			"sourcemod"
		"user"				"root"
		"pass"				""
		//"timeout"			"0"
		//"port"			"0"
	}
	
	"storage-local"
	{
		"driver"			"sqlite"
		"database"			"sourcemod-local"
	}

	"clientprefs"
	{
		"driver"			"sqlite"
		"host"				"localhost"
		"database"			"clientprefs-sqlite"
		"user"				"root"
		"pass"				""
		//"timeout"			"0"
		//"port"			"0"
	}
"csgo"
{
    "driver"    "mysql" // mysql support only
    "host"      "localhost"
    "database"  "storeze"
    "user"      "root"
    "pass"      "xxxxxxx"
    "port"      "0"
}
}

Is there something wrong in the cfg or something else?

**Comment by Kxnrl** [comment]
https://forums.alliedmods.net/showthread.php?t=165270

**Comment by epicbluffer** [comment]
> https://forums.alliedmods.net/showthread.php?t=165270
Thanks but I am stuck at importing data now as below:

Exception reported: Can not retrieve item.parent from database: no result row

I just dont know what to do. I used the item.txt provided and copied to var/lib/mysql/storeze (my database) and also upload files inside utils under /csgo/

THen I copied items.txt and these 4 phps to var/lib/mysql/storeze , and then what should i do?
How can I import parent data to storeze? using terminal?
I tried cd /var/lib/mysql/storeze
          php insertParent.php but terminal indicated lots of errors
I am newbie to Linux and mysql, php......

"Import parent data to your database. ( exec: php insertParent.php or goto: yourwebserver/insertParent.php )
Import item data to your database. ( exec: php insertItem.php or goto: yourwebserver/insertItem.php )
Check item validate ( exec: php loaditem.php or goto: yourwebserver/loaditem.php )"

就是完全不知道要怎麼樣IMPORT DATA....新手求指導 感謝
